lib/std/math/big/int.zig

```diff
@@ -4201,6 +4229,318 @@ fn fixedIntFromSignedDoubleLimb(A: SignedDoubleLimb, storage: []Limb) Mutable {
     };
 }
 
+fn slicesOverlap(a: []const Limb, b: []const Limb) bool {
+    // there is no overlap if a.ptr + a.len <= b.ptr or b.ptr + b.len <= a.ptr
+    return @intFromPtr(a.ptr + a.len) > @intFromPtr(b.ptr) and @intFromPtr(b.ptr + b.len) > @intFromPtr(a.ptr);
+}
+
 test {
     _ = @import("int_test.zig");
 }
+
+const testing_allocator = std.testing.allocator;
+test "llshl shift by whole number of limb" {
+    const padding = std.math.maxInt(Limb);
+
+    var r: [10]Limb = @splat(padding);
+
+    // arbitrary numbers known to fit ?
+    const A: Limb = @truncate(0xCCCCCCCCCCCCCCCCCCCCCCC);
+    const B: Limb = @truncate(0x22222222222222222222222);
+
+    const data = [2]Limb{ A, B };
+    for (0..9) |i| {
+        @memset(&r, padding);
+        const len = llshl(&r, &data, i * @bitSizeOf(Limb));
+
+        try std.testing.expectEqual(i + 2, len);
+        try std.testing.expectEqualSlices(Limb, &data, r[i .. i + 2]);
+        for (r[0..i]) |x|
+            try std.testing.expectEqual(0, x);
+        for (r[i + 2 ..]) |x|
+            try std.testing.expectEqual(padding, x);
+    }
+}
+
+test llshl {
+    if (limb_bits != 64) return error.SkipZigTest;
+
+    // 1 << 63
+    const left_one = 0x8000000000000000;
+    const maxint: Limb = 0xFFFFFFFFFFFFFFFF;
+
+    // zig fmt: off
+    const cases: Cases = &.{
+        .{0,  &.{0},                               &.{0}},
```

Comment:
I really appreciate the amount of testing this adds, but one suggestion I have is if you restructure the code like this:
```zig
try testOneShiftCase(.{ 0, &.{0}, &.{0} });
try testOneShiftCase(.{ 0, &.{1}, &.{1} });
```
etc., then when a test fails in the future we will actually get a meaningful stack trace that points to the failing test case, instead of pointing to useless loops and large array parameters.

Also please follow the `camelCaseFunctionName` style of the surrounding code.